server/src/main/java/org/elasticsearch/index/mapper/SourceFieldMapper.java

```diff
@@ -235,9 +238,6 @@ private boolean isDefault() {
         @Override
         public SourceFieldMapper build() {
             if (enabled.getValue().explicit()) {
-                if (indexMode != null && indexMode.isSyntheticSourceEnabled()) {
```

Comment:
In theory we can disallow disabling source but i don't know if this adds a lot of value. We warn about it in docs and if you really want to do that maybe you should be able to.

Reply:
Let's prohibit disabled source mode for now? We can maybe reconsider this later. But that is then one less variable to think about.